src/wallet/coinselection.h

```diff
@@ -301,6 +304,18 @@ struct SelectionResult
     /** Total weight of the selected inputs */
     int m_weight{0};
 
+    template<typename T>
+    util::Result<bool> InsertInputs(const T& inputs)
+    {
+        // Store sum of combined input sets to check that the results have no shared UTXOs
+        const size_t expected_count = m_selected_inputs.size() + inputs.size();
+        util::insert(m_selected_inputs, inputs);
+        if (!Assume(m_selected_inputs.size() == expected_count)) {
```

Comment:
In e7a94dd2e32f3f0beb0f172ee741e00c674e380b "wallet: change SelectionResult::Merge assert to error"

I'm not sure that `Assume` here is useful. Even for our tests, getting back an expected error is still sufficient to fail tests, we don't need to kill the program.

Reply:
I added it because atm we are simply not supporting the shared UTXOs merge, so this shouldn't happen in our tests neither (we shouldn't be able to trigger this error using the regular transaction creation process, and if we do, then it's a bad bug that should be fixed).

but could remove it too. I'm not having a strong preference for it neither.

Reply:
I have a noob cpp question. Can we somehow enforce that returned error is not ignored by the calling code?

Reply:
> I have a noob cpp question. Can we somehow enforce that returned error is not ignored by the calling code?

yes, by throwing an exception instead of returning a result error message. Which might not be bad. In this case, could be a `runtime_error`.